Ladybird/Qt/SettingsDialog.cpp

```diff
@@ -35,6 +33,20 @@ SettingsDialog::SettingsDialog(QMainWindow* window)
 
     m_new_tab_page = make<QLineEdit>(this);
     m_new_tab_page->setText(Settings::the()->new_tab_page());
+    QObject::connect(m_new_tab_page, &QLineEdit::textChanged, this, [this] {
+        auto url_string = MUST(ak_string_from_qstring(m_new_tab_page->text()));
+        bool is_valid = URL(url_string).is_valid();
+        if (is_valid) {
+            m_new_tab_page->setStyleSheet("");
+            Settings::the()->set_new_tab_page(m_new_tab_page->text());
+        } else {
+            m_new_tab_page->setStyleSheet("border: 1px solid red;");
+            Settings::the()->set_new_tab_page("about:blank");
```

Comment:
If you type in `https://goog`, this will set the new tab page to `"about:blank"`? That feels strange. I would expect it to simply not set the new tab page at all, and keep the old setting.

Reply:
You are right, that behavior was kind of weird. I have changed it so that it now doesn't change the current value when invalid